packages/util/src/object.ts

```diff
@@ -77,9 +79,9 @@ export class PropertyTransformHandler<T extends object> implements ProxyHandler<
                 // or primitive types that cannot be proxied
                 return value;
             }
-            return new Proxy(value, new PropertyTransformHandler(this.transformProperty, this.proxyIdentity));
+            return new Proxy(value, new PropertyTransformHandler(this.transformProperty, this.transformValue, this.proxyIdentity));
         }
-        return value;
+        return this.transformValue(value);
```

Comment:
Maybe we can allow undefined`this.transformValue` to avoid always running the value through the transformer.

Would be good to measure if there is any performance implication of always running it through a transformer; if there is no performance degradation we can keep it like this otherwise first checking if the transformer is set might work better.

Reply:
Will check this.